cmd/app/grpc.go

```diff
@@ -108,6 +109,10 @@ func (g *grpcServer) startTCPListener() {
 
 func (g *grpcServer) startUnixListener() {
 	go func() {
+		if err := os.RemoveAll(LegacyUnixDomainSocket); err != nil {
+			log.Logger.Fatal(err)
+		}
+
```

Comment:
🤔 I don't think that will work. It seems the `@` in the socket address means it's in the abstract namespace which doesn't belong to a file on the filesystem, so not sure why a readonly filesystem won't let me remove it (yet will let it be created?)
Also, digging in to what `os.RemoveAll` does, leads to this (https://github.com/golang/go/blob/e49e8764553bf510b5d9f6fb38aeec0850ec6672/src/os/file_unix.go#L301):
```
	e := ignoringEINTR(func() error {
		return syscall.Unlink(name)
	})
```
so it seems that it's trying to do an unlink already? And lastly, looking into the `lis.SetUnlinkOnClose(true)` call, it seems it won't unlink due to it being an abstract domain socket (i.e. starting with `@`):
```
func (ln *UnixListener) close() error {
	// The operating system doesn't clean up
	// the file that announcing created, so
	// we have to clean it up ourselves.
	// There's a race here--we can't know for
	// sure whether someone else has come along
	// and replaced our socket name already--
	// but this sequence (remove then close)
	// is at least compatible with the auto-remove
	// sequence in ListenUnix. It's only non-Go
	// programs that can mess us up.
	// Even if there are racy calls to Close, we want to unlink only for the first one.
	ln.unlinkOnce.Do(func() {
		if ln.path[0] != '@' && ln.unlink {
			syscall.Unlink(ln.path)
		}
	})
	return ln.fd.Close()
}
```

I'm not super well versed on the nuances of Unix Domain Sockets so will need to do more digging unless someone else has ideas...

Reply:
OK, this:
https://github.com/golang/go/blob/e49e8764553bf510b5d9f6fb38aeec0850ec6672/src/os/file_unix.go#L307
probably explains why the `os.RemoveAll` fails with RO... Whereas I guess creating the socket in the abstract namespace only needs to create a file descriptor

Reply:
After further review, abstract domain sockets are really a linux-only feature. We should probably wrap your proposed change with a check for `if runtime.GOOS != "linux"`

Reply:
Yeah, so the question for me still remains, how are these being handled on MacOS, as they still don't seem to be bound to a file on the filesystem. And I'd like to get to the bottom of why it's hanging about when it doesn't on Linux, but that might be getting a bit too far off course.

I guess I'm wondering now if:
a) there's any point in "fixing" this since it's the legacy GRPC server, like does it have a future anyway? and
b) if this is intended to be run in containers anyway is MacOS nuance something that we should worry about?

I'll just add the condition you've suggested for now, and at least then if the legacy server goes away so does the unnecessary mess :P

Reply:
> how are these being handled on MacOS
🤦 in the most obvious way, just a file in the local dir... will make the change now